### PR TITLE
orm: add ValidateSequence

### PR DESCRIPTION
**Pull request by orkunkl**
On https://github.com/iov-one/tutorial/blob/master/x/orderbook/msg.go#L94-L104 and starter kit will use validateID so I thought maybe It would be good to have this helper since weave based apps might need this method. If you think ValidateSequence method belongs to somewhere else rather than orm package please let me know where to put 👀

!nochangelog

### CODE REVIEW

File: orm/idgen_bucket.go [hunk 1]
```diff
@@ -49,3 +49,15 @@ type IDGeneratorFunc func(db weave.KVStore, obj CloneableData) ([]byte, error)
 func (i IDGeneratorFunc) NextVal(db weave.KVStore, obj CloneableData) ([]byte, error) {
 	return i(db, obj)
 }
+
+// ValidateSequence returns an error if this is not an 8-byte
+// as expected for orm.IDGenBucket
+func ValidateSequence(id []byte) error {
+	if len(id) == 0 {
+		return errors.Wrap(errors.ErrEmpty, "sequence missing")
```
Comment (ruseinov) [inline orm/idgen_bucket.go]: nit: let's do either "missing sequence" or "sequence is missing"

**Comment by ruseinov** [comment]
I think it looks good, but at the same time it feels awkward that we are not using this helper anywhere within the framework.

`orm` is a good place for it, might be good to move it to a file named `helper.go` though as ids could be generated without idGenBucket too.

File: orm/idgen_bucket.go (hunk 1 above)
Comment (orkunkl) [reply orm/idgen_bucket.go]: `missing sequence` fewer words 👍

**Comment by orkunkl** [comment]
IMO `orm/sequence.go` is better. No need to create a new file

File: orm/helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		},
+	}
+
+	for testName, tc := range cases {
+		t.Run(testName, func(t *testing.T) {
+			err := ValidateSequence(tc.bytes)
+			if !tc.wantErr.Is(err) {
+				t.Fatalf("unexpected error: %+v", err)
+			}
+			if tc.wantErr != nil {
```
Comment (husio) [inline orm/helpers_test.go]: This is not needed. :)

**Comment by orkunkl** [comment]
It would have been nice to have this code in v0.21.0 but I guess I am late 🙄

**Comment by ruseinov** [comment]
@orkunkl when your stuff gets approved normally you have to merge it yourself. This process ensures that PR gets merged by someone who knows what it's about. 

I will merge this now.

**Pull request merged**